addons/website/static/tests/builder/website_builder/carousel_item.test.js

```diff
@@ -84,3 +85,23 @@ test("should disable reordering and navigation buttons when carousel has a singl
     expect("button[title='Move Backward']").toHaveClass("disabled");
     expect("button[title='Move Forward']").toHaveClass("disabled");
 });
+
+test("Should disable reordering buttons for s_image_wall when the first or last image is selected", async () => {
+    // Common steps to setup an image wall snippet.
+    const builder = await setupWebsiteBuilderWithSnippet("s_images_wall");
+    queryAll(":iframe img").forEach((imgEl) => (imgEl.src = dummyBase64Img));
+    builder.getEditor().shared.history.addStep();
+
+    expect(":iframe .o_masonry_col img").toHaveCount(6);
+    await contains(":iframe .o_masonry_col:first-child img:first-child").click();
+    await waitFor("[data-action-value='last'].disabled");
+    expect("[data-action-value='last']").toHaveClass("disabled");
+    await contains(":iframe .o_masonry_col:last-child img:last-child").click();
+    expect("[data-action-value='last']").toHaveClass("disabled");
+
+    await contains(":iframe .o_masonry_col:last-child img:first-child").click();
+    await waitFor("[data-action-value='last'].disabled");
+    expect("[data-action-value='last']").toHaveClass("disabled");
+    await contains(":iframe .o_masonry_col:last-child img:last-child").click();
+    expect("[data-action-value='last']").toHaveClass("disabled");
+});
```

Comment:
move it in `community/addons/website/static/tests/builder/website_builder/image_gallery.test.js`